content/collections.md

```diff
@@ -0,0 +1,287 @@
+
+1. We attach the schema to the namespace of `Lists` directly. This allows us to check things against the schema of a list directly (rather than via inserting into the DB, see below), such as in a form (see the forms article).
+2. We specify that the `name` field of a list must be a string, and must exist.
+3. We specify the `incompleteCount` is a number, which on insertion is set to `0` if not otherwise specified.
+4. We specify that the `userId`, which is optional, must be a string matching an id regular expression.
+
+You can see from this example, that with relatively little code we've managed to restrict the format of a list significantly.
+
+### Checking documents against a schema
+
+Now we have a schema, how do we use it?
+
+The straightforward way to use a schema is by using the `check` package. We can write:
+
+```js
+const list = {
+  name: 'My list',
+  incompleteCount: 3
+};
+
+check(list, Lists.schema);
+```
+
+In this case, as the list is valid according to the schema, the `check()` line will run without problems. If however, we wrote:
+
+```js
+const list = {
+  name: 'My list',
+  incompleteCount: 3,
+  madeUpField: 'this should not be here'
+};
+
+check(list, Lists.schema);
+```
+
+// XXX: this isn't actually the case yet. Change this unless we update simple schema to do so..
+Then the `check()` call will throw a `Meteor.ValidationError` which contains details about what is wrong with the `list` document.
+
+### The `Meteor.ValidationError`
+
+What is a `Meteor.ValidationError` [link to docs]? It's a special error that is used in Meteor to indicate a user-input based error in modifying a collection. Typically, the details on a `ValidationError` are used to mark up a form with information about what a user did wrong. In the "Methods and Forms" article, we'll see more about how this works.
+
+## Designing your data schama
+
+Now you are familiar with the basic API of Simple Schema, it's worth considering a few of the constraints of the Meteor system that can influence the design of your data schema. Although generally speaking you can build a Meteor data schema much like any Mongo data schema, there are some important differences.
+
+The most important consideration is due to the way that DDP communicates documents over the wire. The key thing to realize is that DDP sends changes to documents at the level of document *fields*. What this means is that if you have large and complex subfields on document that change often, DDP can send unnecessary changes over the wire.
+
+For instance, in "pure" Mongo you might design the schema so that each list document had a field called `todos` which was an array of todo items:
+
+```js
+Lists.schema = new SimpleSchema({
+  name: {type: String},
+  todos: {type: [Object]},
+  'todos.$.text': {type: String}
+});
+```
+
+(Notice the new Simple Schema syntax here, that allows you describe an array of objects one field at a time).
```

Comment:
I don't think we should introduce a new feature of simple-schema in the same paragraph as we tell you not to use subfields.